pupil_src/shared_modules/marker_tracker_3d/optimization/optimization_generator.py

```diff
@@ -1,63 +1,42 @@
-import threading
-
-from marker_tracker_3d import utils
-from marker_tracker_3d.optimization.optimization import Optimization
-from marker_tracker_3d.optimization.visibility_graphs import VisibilityGraphs
+from marker_tracker_3d.optimization.bundle_adjustment import BundleAdjustment
+from marker_tracker_3d.optimization.initial_guess import InitialGuess
 
 
 def optimization_generator(recv_pipe):
-    # TODO: background Process only do opt_run
-
-    origin_marker_id = None
-    visibility_graphs = None
-    t1 = None
-    opt = None
-    lock = threading.RLock()
+    bundle_adjustment = None
+    initial_guess = None
 
     while True:
         if recv_pipe.poll(0.001):
             msg, data_recv = recv_pipe.recv()
-
-            if msg == "basic_models":
+            if msg == "camera_model":
                 camera_model = data_recv
-                opt = Optimization(camera_model)
-                visibility_graphs = VisibilityGraphs(
-                    camera_model, origin_marker_id=origin_marker_id
+                bundle_adjustment = BundleAdjustment(camera_model)
+                initial_guess = InitialGuess(camera_model)
+
+            elif msg == "opt":
+                data_for_optimization = data_recv
```

Comment:
If you have a data structure like this tuple, which always contains the same number of elements in the same order, it can make sense to make this a named tuple or a class (a class is necessary if the object needs to be mutable) with the according fields. 
The advantage of that is that you make explicit that the tuple will *always* look like this because the structure of the named tuple/class is static. Otherwise, if you e.g. create such tuples elsewhere in your code as well, it is unclear at first glance if this actually is the exact same tuple or if it is slightly different. You'd have to exactly compare both code sections to see if they are the same. 
But if you call the same class constructor instead at both points in your code, it is immediately obvious that they are similar objects.